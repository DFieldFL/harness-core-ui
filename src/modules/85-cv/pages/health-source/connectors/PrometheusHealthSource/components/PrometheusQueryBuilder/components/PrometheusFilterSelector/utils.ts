import type { MultiSelectOption } from '@wings-software/uicore'

export function updateLabelOfSelectedFilter(
  selectedValue: string,
  item: MultiSelectOption,
  items: MultiSelectOption[]
): MultiSelectOption {
  const currentItem = items.find(listItem => listItem.value === item.value)
  if (currentItem) {
    const splitString = (currentItem.label as string).split(':')[0]
    return { ...currentItem, label: `${splitString}:${selectedValue}` }
  }

  return item
}

export function updateMultiSelectOption(
  newOption: MultiSelectOption,
  currentList: MultiSelectOption[]
): MultiSelectOption[] {
  if (!currentList?.length) {
    return []
  }

  const newOptionIndex = currentList.findIndex(listItem => listItem.value === newOption.value)
  if (newOptionIndex > -1) {
    currentList[newOptionIndex] = newOption
  }

  return Array.from(currentList)
}
