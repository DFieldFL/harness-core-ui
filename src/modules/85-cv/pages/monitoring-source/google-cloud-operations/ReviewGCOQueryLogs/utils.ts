import type { GCOMonitoringSourceInfo } from '../GoogleCloudOperationsMonitoringSourceUtils'
import type { GCOLogsDSConfig, MapGCOLogsQueryToService } from '../MapQueriesToHarnessService/types'

export function transformGCOLogsToDSConfig(setupSource: GCOMonitoringSourceInfo): GCOLogsDSConfig {
  const dsConfig: GCOLogsDSConfig = {
    connectorIdentifier: setupSource.connectorRef?.value as string,
    type: 'STACKDRIVER_LOG',
    accountId: setupSource.accountId,
    projectIdentifier: setupSource.projectIdentifier,
    orgIdentifier: setupSource.orgIdentifier,
    identifier: setupSource.identifier,
    monitoringSourceName: setupSource.name,
    logConfigurations: []
  }

  for (const entry of setupSource.mappedServicesAndEnvs.entries()) {
    const {
      metricName,
      query,
      serviceIdentifier,
      envIdentifier,
      serviceInstance,
      messageIdentifier
    }: MapGCOLogsQueryToService = entry[1]

    dsConfig.logConfigurations.push({
      envIdentifier: envIdentifier?.value as string,
      serviceIdentifier: serviceIdentifier?.value as string,
      logDefinition: {
        name: metricName,
        query,
        serviceInstanceIdentifier: serviceInstance,
        messageIdentifier: messageIdentifier
      }
    })
  }

  return dsConfig
}
