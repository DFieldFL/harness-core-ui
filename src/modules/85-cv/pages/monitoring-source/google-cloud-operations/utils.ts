import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import {
  GCODSConfig,
  GCOMonitoringSourceInfo,
  buildGCOMonitoringSourceInfo,
  GCOProduct
} from './GoogleCloudOperationsMonitoringSourceUtils'
import type { GCOLogsDSConfig } from './MapQueriesToHarnessService/types'

export function transformGetResponseForStackDriver(
  gcoConfig: GCODSConfig,
  params: ProjectPathProps & { identifier: string }
): GCOMonitoringSourceInfo {
  const gcoInfo: GCOMonitoringSourceInfo = buildGCOMonitoringSourceInfo(params, GCOProduct.CLOUD_METRICS)
  if (!gcoConfig) {
    return gcoConfig
  }

  gcoInfo.accountId = gcoConfig.accountId
  gcoInfo.identifier = gcoConfig.identifier
  gcoInfo.name = gcoConfig.monitoringSourceName
  gcoInfo.type = 'STACKDRIVER'
  gcoInfo.orgIdentifier = gcoConfig.orgIdentifier
  gcoInfo.projectIdentifier = gcoConfig.projectIdentifier
  gcoInfo.isEdit = !!gcoConfig.identifier
  gcoInfo.connectorRef = {
    label: gcoConfig.connectorIdentifier as string,
    value: gcoConfig.connectorIdentifier as string
  }
  gcoInfo.product = GCOProduct.CLOUD_METRICS
  gcoInfo.selectedDashboards = []
  gcoInfo.selectedMetrics = new Map()

  for (const config of gcoConfig.metricConfigurations) {
    const { envIdentifier, serviceIdentifier, metricDefinition } = config
    // TODO: Remove this once backend fixes discripancy in key names manualQuery and isManualQuery
    const manualQuery = metricDefinition?.isManualQuery || (metricDefinition as any)?.manualQuery
    if (
      !envIdentifier ||
      !serviceIdentifier ||
      !metricDefinition ||
      !metricDefinition.dashboardName ||
      !(manualQuery ? true : metricDefinition.dashboardPath) ||
      !metricDefinition.metricName
    )
      continue

    const metricTags: any = {}
    for (const tag of metricDefinition.metricTags || []) {
      metricTags[tag] = ''
    }

    gcoInfo.selectedDashboards.push({ name: metricDefinition.dashboardName, path: metricDefinition.dashboardPath })
    gcoInfo.selectedMetrics.set(metricDefinition.metricName, {
      metricName: metricDefinition.metricName,
      metricTags,
      dashboardName: metricDefinition.dashboardName,
      dashboardPath: metricDefinition.dashboardPath,
      service: { label: serviceIdentifier, value: serviceIdentifier },
      environment: { label: envIdentifier, value: envIdentifier },
      isManualQuery: manualQuery,
      riskCategory: `${metricDefinition.riskProfile?.category}/${metricDefinition.riskProfile?.metricType}`,
      higherBaselineDeviation: metricDefinition.riskProfile?.thresholdTypes?.includes('ACT_WHEN_HIGHER'),
      lowerBaselineDeviation: metricDefinition.riskProfile?.thresholdTypes?.includes('ACT_WHEN_LOWER'),
      query: JSON.stringify(metricDefinition.jsonMetricDefinition, null, 2)
    })
  }

  return gcoInfo
}

export function transformGetResponseForStackDriverLogs(
  gcoLogsConfig: GCOLogsDSConfig,
  params: ProjectPathProps & { identifier: string }
): GCOMonitoringSourceInfo {
  const gcoLogsInfo: GCOMonitoringSourceInfo = buildGCOMonitoringSourceInfo(params, GCOProduct.CLOUD_LOGS)
  if (!gcoLogsConfig) {
    return gcoLogsConfig
  }

  gcoLogsInfo.accountId = gcoLogsConfig.accountId
  gcoLogsInfo.identifier = gcoLogsConfig.identifier
  gcoLogsInfo.name = gcoLogsConfig.monitoringSourceName
  gcoLogsInfo.type = 'STACKDRIVER_LOG'
  gcoLogsInfo.orgIdentifier = gcoLogsConfig.orgIdentifier
  gcoLogsInfo.projectIdentifier = gcoLogsConfig.projectIdentifier
  gcoLogsInfo.connectorRef = {
    label: gcoLogsConfig.connectorIdentifier as string,
    value: gcoLogsConfig.connectorIdentifier as string
  }
  gcoLogsInfo.product = GCOProduct.CLOUD_LOGS
  gcoLogsInfo.isEdit = !!gcoLogsInfo.identifier

  for (const config of gcoLogsConfig.logConfigurations) {
    const { envIdentifier, serviceIdentifier, logDefinition } = config
    if (
      !envIdentifier ||
      !serviceIdentifier ||
      !logDefinition ||
      !logDefinition.name ||
      !logDefinition.query ||
      !logDefinition.serviceInstanceIdentifier ||
      !logDefinition.messageIdentifier
    ) {
      continue
    }

    gcoLogsInfo.mappedServicesAndEnvs.set(logDefinition.name, {
      metricName: logDefinition.name,
      serviceIdentifier: { label: serviceIdentifier, value: serviceIdentifier },
      envIdentifier: { label: envIdentifier, value: envIdentifier },
      serviceInstance: logDefinition.serviceInstanceIdentifier,
      messageIdentifier: logDefinition.messageIdentifier,
      query: logDefinition.query
    })
  }

  return gcoLogsInfo
}
