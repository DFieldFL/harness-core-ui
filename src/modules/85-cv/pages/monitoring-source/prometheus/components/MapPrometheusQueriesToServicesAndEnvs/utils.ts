import type { SelectOption } from '@wings-software/uicore'
import { isNumber } from 'lodash-es'
import type { FormikProps } from 'formik'
import type { UseStringsReturn } from 'framework/strings'
import type { MapPrometheusQueryToService } from '../../constants'
import { MapPrometheusQueryToServiceFieldNames } from './constants'

type UpdateSelectedMetricsMap = {
  updatedMetric: string
  oldMetric: string
  mappedMetrics: Map<string, MapPrometheusQueryToService>
  formikProps: FormikProps<MapPrometheusQueryToService | undefined>
}

export function updateSelectedMetricsMap({
  updatedMetric,
  oldMetric,
  mappedMetrics,
  formikProps
}: UpdateSelectedMetricsMap) {
  const updatedMap = new Map(mappedMetrics)

  // in the case where user updates metric name, update the key for current value
  if (oldMetric !== formikProps.values?.metricName) {
    updatedMap.delete(oldMetric)
  }

  // if newly created metric create form object
  if (!updatedMap.has(updatedMetric)) {
    updatedMap.set(updatedMetric, { metricName: updatedMetric, query: '', isManualQuery: false })
  }

  // update map with current form data
  if (formikProps.values?.metricName) {
    updatedMap.set(formikProps.values.metricName, formikProps.values as MapPrometheusQueryToService)
  }
  return { selectedMetric: updatedMetric, mappedMetrics: updatedMap }
}

export function validateMappings(
  getString: UseStringsReturn['getString'],
  createdMetrics: string[],
  selectedMetricIndex: number,
  values?: MapPrometheusQueryToService
): { [fieldName: string]: string } {
  const requiredFieldErrors = {
    [MapPrometheusQueryToServiceFieldNames.ENVIRONMENT]: getString('cv.monitoringSources.envValidation'),
    [MapPrometheusQueryToServiceFieldNames.ENVIRONMENT_FILTER]: getString(
      'cv.monitoringSources.prometheus.validation.filterOnEnvironment'
    ),
    [MapPrometheusQueryToServiceFieldNames.METRIC_NAME]: getString('cv.monitoringSources.metricNameValidation'),
    [MapPrometheusQueryToServiceFieldNames.PROMETHEUS_METRIC]: getString(
      'cv.monitoringSources.prometheus.validation.promethusMetric'
    ),
    [MapPrometheusQueryToServiceFieldNames.RISK_CATEGORY]: getString(
      'cv.monitoringSources.gco.mapMetricsToServicesPage.validation.riskCategory'
    ),
    [MapPrometheusQueryToServiceFieldNames.SERVICE]: getString('cv.monitoringSources.serviceValidation'),
    [MapPrometheusQueryToServiceFieldNames.SERVICE_FILTER]: getString(
      'cv.monitoringSources.prometheus.validation.filterOnService'
    ),
    [MapPrometheusQueryToServiceFieldNames.SERVICE_INSTANCE]: getString(
      'cv.monitoringSources.prometheus.validation.serviceInstanceIdentifier'
    ),
    [MapPrometheusQueryToServiceFieldNames.QUERY]: getString(
      'cv.monitoringSources.gco.manualInputQueryModal.validation.query'
    )
  }

  if (!values) {
    return requiredFieldErrors
  }

  if (values.isManualQuery) {
    delete requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.SERVICE_FILTER]
    delete requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.ENVIRONMENT_FILTER]
    delete requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.SERVICE_INSTANCE]
    delete requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.PROMETHEUS_METRIC]
  }

  for (const fieldName of Object.keys(requiredFieldErrors)) {
    switch (fieldName) {
      case MapPrometheusQueryToServiceFieldNames.ENVIRONMENT_FILTER:
        if (values.envFilter?.length) delete requiredFieldErrors[fieldName]
        break
      case MapPrometheusQueryToServiceFieldNames.SERVICE_FILTER:
        if (values.serviceFilter?.length) delete requiredFieldErrors[fieldName]
        break
      default:
        if ((values as any)[fieldName]) delete requiredFieldErrors[fieldName]
    }
  }

  if (values.lowerBaselineDeviation !== true && values.higherBaselineDeviation !== true) {
    requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.LOWER_BASELINE_DEVIATION] = getString(
      'cv.monitoringSources.prometheus.validation.deviation'
    )
  }

  const duplicateNames = createdMetrics?.filter((metricName, index) => {
    if (index === selectedMetricIndex) {
      return false
    }
    return metricName === values.metricName
  })

  if (values.metricName && duplicateNames.length) {
    requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.METRIC_NAME] = getString(
      'cv.monitoringSources.prometheus.validation.metricNameUnique'
    )
  }

  if (
    !requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.QUERY] &&
    isNumber(values.recordCount) &&
    values.recordCount > 1
  ) {
    requiredFieldErrors[MapPrometheusQueryToServiceFieldNames.QUERY] = getString(
      'cv.monitoringSources.prometheus.validation.recordCount'
    )
  }

  return requiredFieldErrors
}

export function initializePrometheusGroupNames(
  mappedMetrics: Map<string, MapPrometheusQueryToService>,
  getString: UseStringsReturn['getString']
): SelectOption[] {
  const groupNames = Array.from(mappedMetrics?.entries())
    .map(metric => {
      const { groupName } = metric?.[1] || {}
      return groupName || null
    })
    .filter(groupItem => groupItem !== null) as SelectOption[]
  return [{ label: getString('cv.addNew'), value: '' }, ...groupNames]
}
