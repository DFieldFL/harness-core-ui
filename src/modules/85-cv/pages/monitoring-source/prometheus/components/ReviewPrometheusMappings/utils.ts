import type { MultiSelectOption } from '@wings-software/uicore'
import type { MetricDefinition, TimeSeriesMetricDefinition } from 'services/cv'

import type {
  PrometheusDSConfig,
  PrometheusSetupSource,
  MapPrometheusQueryToService,
  RiskProfileCatgory,
  PrometheusFilter
} from '../../constants'

export function transformLabelToPrometheusFilter(options?: MultiSelectOption[]): PrometheusFilter[] {
  const filters: PrometheusFilter[] = []
  if (!options?.length) {
    return filters
  }

  for (const option of options) {
    const value = option.label.split(':')
    filters.push({ labelName: value[0], labelValue: value[1] })
  }

  return filters
}

export function transformPrometheusSourceToDSConfig(setupSource: PrometheusSetupSource): PrometheusDSConfig {
  const dsConfig: PrometheusDSConfig = {
    connectorIdentifier: setupSource.connectorRef?.value,
    type: 'PROMETHEUS',
    accountId: setupSource.accountId,
    projectIdentifier: setupSource.projectIdentifier,
    orgIdentifier: setupSource.orgIdentifier,
    productName: setupSource.productName,
    identifier: setupSource.identifier,
    monitoringSourceName: setupSource.monitoringSourceName,
    metricDefinitions: []
  }

  for (const entry of setupSource.mappedServicesAndEnvs.entries()) {
    const {
      envIdentifier,
      envFilter,
      metricName,
      groupName,
      prometheusMetric,
      serviceFilter,
      serviceInstance,
      serviceIdentifier,
      isManualQuery,
      query,
      aggregator,
      additionalFilter,
      riskCategory,
      lowerBaselineDeviation,
      higherBaselineDeviation
    }: MapPrometheusQueryToService = entry[1]

    if (!envIdentifier || !serviceIdentifier || !groupName || !metricName || !riskCategory) {
      continue
    }

    if (!isManualQuery && (!prometheusMetric || !envFilter || !serviceFilter || !serviceInstance)) {
      continue
    }

    const [category, metricType] = riskCategory?.split('/') || []
    const thresholdTypes: TimeSeriesMetricDefinition['thresholdType'][] = []
    if (lowerBaselineDeviation) {
      thresholdTypes.push('ACT_WHEN_LOWER')
    }
    if (higherBaselineDeviation) {
      thresholdTypes.push('ACT_WHEN_HIGHER')
    }

    dsConfig.metricDefinitions.push({
      prometheusMetric,
      metricName,
      serviceFilter: transformLabelToPrometheusFilter(serviceFilter),
      isManualQuery,
      query,
      envFilter: transformLabelToPrometheusFilter(envFilter),
      additionalFilters: transformLabelToPrometheusFilter(additionalFilter),
      aggregation: aggregator,
      serviceInstanceFieldName: serviceInstance,
      groupName: groupName.value as string,
      envIdentifier: envIdentifier.value as string,
      serviceIdentifier: serviceIdentifier.value as string,
      riskProfile: {
        category: category as RiskProfileCatgory,
        metricType: metricType as MetricDefinition['type'],
        thresholdTypes
      }
    })
  }

  return dsConfig
}
