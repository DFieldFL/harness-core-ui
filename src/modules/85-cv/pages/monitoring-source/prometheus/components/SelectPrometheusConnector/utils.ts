import * as Yup from 'yup'
import type { DefineYourMonitoringSourceProps } from '@cv/components/CVSetupSourcesView/DefineYourMonitoringSource/DefineYourMonitoringSource'
import type { UseStringsReturn } from 'framework/strings'
import { StringUtils } from '@common/exports'

export function getDefineMonitoringSourceBaseProps(
  getString: UseStringsReturn['getString']
): Omit<DefineYourMonitoringSourceProps, 'mainHeading' | 'subHeading' | 'formikProps'> {
  return {
    stepLabelProps: {
      stepNumber: 1,
      totalSteps: 3
    },
    iconLabel: getString('connectors.prometheusLabel'),
    sourceIcon: { name: 'service-prometheus' }
  }
}

export function getValidationSchema(getString: UseStringsReturn['getString']): Record<string, any> {
  return Yup.object({
    monitoringSourceName: Yup.string().required(getString('cv.onboarding.selectProductScreen.validationText.name')),
    identifier: Yup.string().when('name', {
      is: val => val?.length,
      then: Yup.string()
        .trim()
        .required(getString('cv.onboarding.selectProductScreen.validationText.identifier'))
        .matches(
          /^(?![0-9])[0-9a-zA-Z_$]*$/,
          getString('cv.onboarding.selectProductScreen.validationText.validIdRegex')
        )
        .notOneOf(StringUtils.illegalIdentifiers)
    }),
    connectorRef: Yup.object().required(getString('cv.onboarding.selectProductScreen.validationText.connectorRef')),
    productName: Yup.string().required(getString('cv.onboarding.selectProductScreen.validationText.product'))
  })
}
