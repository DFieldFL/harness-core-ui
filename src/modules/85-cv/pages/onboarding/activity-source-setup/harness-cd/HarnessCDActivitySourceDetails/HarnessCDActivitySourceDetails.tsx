import React from 'react'
import { Container, Formik, FormikForm } from '@wings-software/uicore'
import * as Yup from 'yup'
import { useHistory, useParams } from 'react-router-dom'
import SyncStepDataValues from '@cv/utils/SyncStepDataValues'
import routes from '@common/RouteDefinitions'
import { StringUtils } from '@common/exports'
import { SubmitAndPreviousButtons } from '@cv/pages/onboarding/SubmitAndPreviousButtons/SubmitAndPreviousButtons'
import { useStrings } from 'framework/strings'
import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import ActivitySourceDetails from '../../ActivitySourceDetails/ActivitySourceDetails'

export interface HarnessCDActivitySourceDetailsProps {
  initialValues?: any
  onSubmit?: (data: any) => void
}

const HarnessCDActivitySourceDetails: React.FC<HarnessCDActivitySourceDetailsProps> = props => {
  const { getString } = useStrings()
  const { initialValues } = props
  const history = useHistory()
  const { projectIdentifier, orgIdentifier, accountId, activitySourceId } = useParams<
    ProjectPathProps & { activitySourceId: string }
  >()
  return (
    <Container style={{ position: 'relative', top: 80 }}>
      <Formik
        initialValues={initialValues}
        validationSchema={Yup.object().shape({
          name: Yup.string().trim().required(getString('validation.nameRequired')),
          identifier: Yup.string().when('name', {
            is: val => val?.length,
            then: Yup.string()
              .trim()
              .required(getString('validation.identifierRequired'))
              .matches(/^(?![0-9])[0-9a-zA-Z_$]*$/, getString('validation.validIdRegex'))
              .notOneOf(StringUtils.illegalIdentifiers)
          })
        })}
        onSubmit={values => {
          props.onSubmit?.(values)
        }}
      >
        {formik => {
          return (
            <FormikForm>
              <ActivitySourceDetails
                heading={getString('cv.activitySources.harnessCD.select')}
                iconLabel={getString('cv.activitySources.harnessCD.iconLabel')}
                iconName={'cd-main'}
                isEditMode={!!activitySourceId}
              />
              <SubmitAndPreviousButtons
                onPreviousClick={() =>
                  history.push(
                    routes.toCVAdminSetup({
                      projectIdentifier: projectIdentifier as string,
                      orgIdentifier: orgIdentifier as string,
                      accountId
                    })
                  )
                }
                onNextClick={() => {
                  formik.submitForm()
                }}
              />
              <SyncStepDataValues values={formik.values} listenToValues={initialValues} onUpdate={formik.setValues} />
            </FormikForm>
          )
        }}
      </Formik>
    </Container>
  )
}

export default HarnessCDActivitySourceDetails
