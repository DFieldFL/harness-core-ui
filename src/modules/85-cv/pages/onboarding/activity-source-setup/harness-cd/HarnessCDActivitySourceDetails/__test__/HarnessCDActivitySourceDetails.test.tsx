import React from 'react'
import { render, act, fireEvent } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { InputTypes, setFieldValue } from '@common/utils/JestFormHelper'
import HarnessCDActivitySourceDetails from '../HarnessCDActivitySourceDetails'

describe('HarnessCDActivitySourceDetails snapshot test', () => {
  test('initializes ok', async () => {
    const { container } = render(
      <TestWrapper pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}>
        <HarnessCDActivitySourceDetails initialValues={{}} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('onSubmit is called', async () => {
    const onSubmit = jest.fn()
    const { container } = render(
      <TestWrapper pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}>
        <HarnessCDActivitySourceDetails initialValues={{}} onSubmit={onSubmit} />
      </TestWrapper>
    )

    await setFieldValue({
      container,
      type: InputTypes.TEXTFIELD,
      fieldId: 'name',
      value: 'TestName'
    })

    await act(async () => {
      fireEvent.click(container.querySelector('button[type="submit"]')!)
    })

    expect(onSubmit).toHaveBeenCalledWith({
      identifier: 'TestName',
      name: 'TestName'
    })
  })
})
