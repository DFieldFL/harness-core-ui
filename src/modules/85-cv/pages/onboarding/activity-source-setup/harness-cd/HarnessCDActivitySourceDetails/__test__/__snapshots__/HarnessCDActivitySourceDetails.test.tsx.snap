// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HarnessCDActivitySourceDetails snapshot test initializes ok 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
    style="position: relative; top: 80px;"
  >
    <div
      class="OverlaySpinner--overlaySpinner"
    >
      <form
        class="FormikForm--main"
      >
        <div>
          <div
            class="StyledProps--font StyledProps--main mainDetails"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--font-size-medium StyledProps--margin-top-large StyledProps--margin-bottom-large"
            >
              cv.activitySources.harnessCD.select
            </p>
            <div
              class="Layout--vertical Layout--layout-spacing-small StyledProps--main monitoringCard"
            >
              <div
                class="bp3-card bp3-interactive bp3-elevation-0 Card--card selectionCard selectedCard Card--selected Card--interactive"
                tabindex="0"
              >
                <div
                  class="triangle"
                >
                  <span
                    class="bp3-icon bp3-icon-tick StyledProps--main tick StyledProps--color StyledProps--color-white"
                    icon="tick"
                  >
                    <svg
                      data-icon="tick"
                      height="14"
                      viewBox="0 0 16 16"
                      width="14"
                    >
                      <desc>
                        tick
                      </desc>
                      <path
                        d="M14 3c-.28 0-.53.11-.71.29L6 10.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l8-8A1.003 1.003 0 0014 3z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </div>
                <div
                  class="Card--cardIcon cardIconSelected"
                >
                  <span
                    class="bp3-icon StyledProps--main"
                    data-icon="harness"
                  >
                    <svg
                      height="40"
                      viewBox="0 0 25 25"
                      width="40"
                    >
                      <path
                        d="M22.974 8.186L16.708 1.88a7.023 7.023 0 00-2.59-1.572C11.98-.409 9.858.154 8.135 1.88L1.868 8.186a7.081 7.081 0 00-1.562 2.606c-.713 2.153-.153 4.289 1.562 6.023l6.266 6.306a7.113 7.113 0 002.589 1.58 5.663 5.663 0 001.808.308c1.503 0 2.946-.64 4.177-1.88l6.266-6.305a7.08 7.08 0 001.562-2.606c.713-2.162.153-4.298-1.562-6.032zm-10.07-4.477c.706.213 1.232.7 1.274.735l1.851 1.862-3.608 3.631-3.608-3.63 1.859-1.872c.5-.496 1.197-1.034 2.233-.726zm-9.219 8.305c.212-.71.696-1.24.73-1.282l1.85-1.863 3.609 3.632-3.608 3.63-1.86-1.87c-.492-.504-1.027-1.214-.721-2.247zm8.252 9.279c-.705-.214-1.231-.701-1.273-.735l-1.851-1.863 3.608-3.631 3.608 3.631-1.86 1.871c-.5.496-1.205 1.034-2.232.726zm9.22-8.305c-.212.709-.696 1.239-.73 1.281l-1.851 1.863-3.608-3.631 3.608-3.632 1.86 1.872c.492.504 1.027 1.204.72 2.247z"
                        fill="#4facdf"
                      />
                    </svg>
                  </span>
                </div>
              </div>
              <p
                class="StyledProps--font StyledProps--main cardLabel StyledProps--color StyledProps--color-black"
              >
                cv.activitySources.harnessCD.iconLabel
              </p>
            </div>
            <div
              class="StyledProps--font StyledProps--main main"
            >
              <div
                class="StyledProps--font StyledProps--main connectorFormNameWrapper"
              >
                <div
                  class="StyledProps--font StyledProps--main connectorFormNameElm"
                >
                  <div
                    class="InputWithIdentifier--txtNameContainer TextInput--main"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main InputWithIdentifier--txtIdContainer"
                    >
                      <p
                        class="StyledProps--font StyledProps--main"
                      >
                        ID
                        :
                      </p>
                      <span
                        class="bp3-popover-wrapper"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <div
                            class="bp3-editable-text bp3-editable-text-placeholder InputWithIdentifier--idValue"
                            tabindex="0"
                          >
                            <span
                              class="bp3-editable-text-content"
                              style="height: 1px; line-height: 1px; min-width: 0;"
                            />
                          </div>
                        </span>
                      </span>
                      <span
                        class="bp3-icon bp3-icon-edit StyledProps--main"
                        icon="edit"
                        style="vertical-align: middle;"
                      >
                        <svg
                          data-icon="edit"
                          height="12"
                          viewBox="0 0 16 16"
                          width="12"
                        >
                          <desc>
                            edit
                          </desc>
                          <path
                            d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </div>
                    <div
                      class="bp3-form-group"
                      data-id="name-0"
                    >
                      <label
                        class="bp3-label"
                        for="name"
                      >
                        cv.activitySources.name
                         
                        <span
                          class="bp3-text-muted"
                        />
                      </label>
                      <div
                        class="bp3-form-content"
                      >
                        <div
                          class="bp3-input-group"
                        >
                          <input
                            autocomplete="off"
                            class="bp3-input"
                            name="name"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main"
                  style="justify-content: center;"
                >
                  <p
                    class="StyledProps--font StyledProps--main fieldToggleLabel StyledProps--color StyledProps--color-primary7"
                  >
                    + Description
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main fieldToggleLabel StyledProps--color StyledProps--color-primary7"
                  >
                    + Tags
                  </p>
                </div>
              </div>
              <div
                class="StyledProps--font StyledProps--main connectorFormElm"
              />
            </div>
          </div>
          <div
            class="Layout--horizontal StyledProps--main main"
          >
            <button
              class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color"
              type="button"
            >
              <span
                class="bp3-icon bp3-icon-chevron-left StyledProps--main StyledProps--padding-right-xsmall"
                icon="chevron-left"
              >
                <svg
                  data-icon="chevron-left"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    chevron-left
                  </desc>
                  <path
                    d="M7.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71l-4 4C5.11 7.47 5 7.72 5 8c0 .28.11.53.29.71l4 4a1.003 1.003 0 001.42-1.42L7.41 8z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <span
                class="bp3-button-text"
              >
                previous
              </span>
            </button>
            <button
              class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
              type="submit"
            >
              <span
                class="bp3-button-text"
              >
                next
              </span>
              <span
                class="bp3-icon bp3-icon-chevron-right StyledProps--main"
                icon="chevron-right"
              >
                <svg
                  data-icon="chevron-right"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    chevron-right
                  </desc>
                  <path
                    d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
